src/Networking/websocket/WebSocketProtocol.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <functional>
+#include <map>
+#include <string>
+
+#include <nlohmann/json.hpp>
+#include <websocketpp/config/asio_no_tls.hpp>
+#include <websocketpp/server.hpp>
+
+namespace websocket {
+
+using nlohmann::json;
+
+/**
+ * @brief Defines a protocol which will be served at an endpoint of a server.
+ * This protocol defines several message types, and processes messages of those types,
+ * dispatching events to message handlers wherever necessary.
+ */
+class WebSocketProtocol {
+public:
+	WebSocketProtocol(const std::string& protocolPath);
```

Comment:
This should be documented - in particular, what is the expected format of `protocolPath`? Should it have slashes (leading, trailing)?